array-methods.js

```diff
@@ -83,8 +139,22 @@ var sumOfInterests = null;
     round this number to the nearest 10th of a cent before moving on.
   )
  */
-var stateSums = null;
+var stateSums = {};
 
+function groupStates(element) {
+  let balance = parseFloat(element.amount);
+  let state = element.state;
+
+  if (!stateSums[state]) {
+    stateSums[state] = 0.0
+  }
+
+  stateSums[state] += balance;
+  stateSums[state] = Math.round(stateSums[state] * 100) / 100
+  return stateSums;
+}
+
+dataset.bankBalances.forEach(groupStates);
```

Comment:
Try not to use forEach here since this will always return undefined. Instead try to use map and have stateSums equal the result of that map function.